core/src/main/java/org/infinispan/interceptors/impl/ClusteringInterceptor.java

```diff
@@ -114,6 +114,9 @@ public synchronized boolean completeExceptionally(Throwable ex) {
       @Override
       public Object apply(InvocationContext rCtx, VisitableCommand rCommand, Object rv) throws Throwable {
          assert rv == null; // value with which the allFuture has been completed
+         if (hasUnsureResponse) {
```

Comment:
@danberindei I think I've screwed this one as well; Here should be the check for lost data as well.